pkg/solana/chainreader/batch.go

```diff
@@ -154,6 +158,35 @@ func doMethodBatchCall(ctx context.Context, lggr logger.Logger, client MultipleA
 			continue
 		}
 
+		// HACK: workaround for OffRampLatestConfigDetails: we need to use an input param to filter an array in the return values, not for seeds
+		if batchCall.ReadName == ccipconsts.MethodNameOffRampLatestConfigDetails {
+			params := batchCall.Params.(map[string]any)
+			ocrPluginType := params["ocrPluginType"].(uint8)
+
+			output := map[string]any{}
+			err := asValueDotValue(
+				ctx,
+				rBinding,
+				&output,
+				wrapDecodeValuer(rBinding, data[dataIdx]),
+			)
+			if err != nil {
+				results[idx].err = err
+				continue
+			}
+
+			config := reflect.Indirect(reflect.ValueOf(output["Ocr3"])).Index(int(ocrPluginType)).Interface()
+			output["OCRConfig"] = config
+
+			// weakdecode so the uint8 field is cast to bool
+			err = mapstructure.WeakDecode(output, results[dataIdx].returnVal)
```

Comment:
This should probably be the following:

```
err = mapstructure.WeakDecode(output, results[idx].returnVal)
```

`results` should be the same length as `batch` therefore the indexes should match.

Reply:
Just pushed a fix